api/routes/router.go

```diff
@@ -31,6 +32,7 @@ func InitRouter(r *gin.Engine) {
 func initRoutes() {
 	setGetPostRoutes("/hello", getGreetings, postGreetings)
 	setGetPostRoutes("/executive", getExecutive, postExecutive)
+	setGetPostRoutes("/executive/:department", getDepartmentExecutive, postExecutive)
```

Comment:
we don't need the postExecutive for /:department because POST /executive handles it. 

we can have getDepartmentExecutive only for this line and use `router.GET(route)` instead of `setGetPostRoutes`

Reply:
changed! thanks